src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -190,7 +190,6 @@ static inline void
 sol_str_slice_copy(char *dst, const struct sol_str_slice src)
 {
     memcpy(dst, src.data, src.len);
-    dst[src.len] = 0;
```

Comment:
this is not invalid, when you copy from a slice to a `char*` (string), it is expected to have space for the contents AND the trailing nul byte. Maybe this needs documentation to be enhanced?

eventually using `sol_buffer` is a better approach as it can choose to use nul-byte or not.